src/transformStateWithClones.js

```diff
@@ -8,6 +8,30 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const stateHistory = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
```

Comment:
Use destruction for `action`
```suggestion
  for (const action of actions) {
    const { type, ........ } = action;
    
    switch (type) {
```